controlplane/kubeadm/controllers/controller.go

```diff
@@ -282,17 +282,21 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return ctrl.Result{}, nil
 	}
 
-	controlPlane := internal.NewControlPlane(cluster, kcp, ownedMachines)
+	controlPlane, err := internal.NewControlPlane(ctx, r.Client, cluster, kcp, ownedMachines)
+	if err != nil {
+		logger.Error(err, "failed to initialize control plane")
```

Comment:
```suggestion
```

The error is logged by controller-runtime, do we need to double-log here?

Reply:
I sort of wish we could disable the controller-runtime logs, I think they're less helpful
> status.go:47] controllers/KubeadmControlPlane "msg"="failed to initialize control plane" "error"="oops" "cluster"="bmo" "kubeadmControlPlane"="bmo-control-plane" "namespace"="default"

vs

> controller.go:248] controller-runtime/controller "msg"="Reconciler error" "error"="oops" "controller"="kubeadmcontrolplane" "name"="bmo-control-plane" "namespace"="default"

I've often been confused by `controller.go:248` being the stack trace, since that ends up being [the line from controller-runtime](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.5.9/pkg/internal/controller/controller.go#L248), not our code. We also get a somewhat more helpful message and any additional context we have put on our logger instance.

Reply:
+1 to double logging. When C-R logs, because it loses the provenance of the log message, it's essentially useless.

What I did for Velero, which uses Logrus directly and not a logging abstraction interface, is to add 2 logging hooks: 1 that records the full path and line number where the log message was created (just the file basename is often ambiguous), and a 2nd that records the full path and line number where the error message was created (assuming it has stack trace information courtesy of pkg/errors).

I know we can't do the exact same thing here, as we're not using Logrus, but it would be nice to find an equivalent.

Reply:
Thanks all, it'd be great to capture this feedback within this PR in controller-runtime https://github.com/kubernetes-sigs/controller-runtime/pull/1054/files where we're now plowing through a context with a logger and preset values, we should improve these as best we can

Reply:
https://github.com/kubernetes-sigs/controller-runtime/pull/1054/files#r463657933